source/engineering/abaqus/multi-node.md

```diff
@@ -0,0 +1,61 @@
+---
+title: "Multi-Node Jobs"
+---
+
+Large parallel Abaqus jobs, using more than 28 processors, can be run on across multiple nodes using MPI.
+
+```{tip}
+Multi-node parallelism is best-suited to Abaqus/Explicit.
+Abaqus/Standard (implicit) does not scale well across multiple nodes
+```
+
+## Job Script Template
+
+Here is an example job submission script for a multi-node Abaqus job:
+
+```{code-block} bash
+---
+linenos: true
+---
+#!/usr/bin/env bash -l
+# 
+#SBATCH --job-name=my_job
+#SBATCH --nodes=2
+#SBATCH --ntasks-per-node=28
+#SBATCH --cpus-per-task=1 
+#SBATCH --time=0:10:00 
+#SBATCH --mem-per-cpu=4000M
+
+# Load modules 
+module load apps/abaqus/2018
+# module load intel/2016b                          # BlueCrystal (Phase 4)
+# module load lang/intel-parallel-studio-xe/2020   # BluePebble
+
+# Unset SLURM's Global Task ID for ABAQUS's PlatformMPI to work 
+unset SLURM_GTIDS 
+
+# Get allocated nodes for Abaqus
+env_file=abaqus_v6.env 
+node_list=$(scontrol show hostname ${SLURM_NODELIST} | sort -u) 
+mp_host_list="[" 
+for host in ${node_list}; do 
+    mp_host_list="${mp_host_list}['$host', ${SLURM_CPUS_ON_NODE}]," 
+done 
+mp_host_list=$(echo ${mp_host_list} | sed -e "s/,$/]/") 
+echo "mp_host_list=${mp_host_list}"  >> ${env_file} 

```

Comment:
Yes, the `env_file` may already exist with other configuration options in it which we don't want to overwrite.

Reply:
That makes sense. However, does this mean that if you run the script twice in the same location without replacing the `env_file`, then you will get two `mp_host_list=[...]` lines? Hopefully Abaqus will use the last line in the file.

It looks like this is true according to [abaqus-docs.mit.edu](https://abaqus-docs.mit.edu/2017/English/SIMACAEILGRefMap/simailg-m-EnvUsing-sb.htm):

> If the same parameter is defined in more than one of these environment files or is defined more than once within a file, the last definition encountered will be used.

Reply:
Ah I see what you were getting at. Yes the `env_file` is essentially a python script that gets sourced within Abaqus, so that isn't an issue. Ideally, researchers shouldn't be launching jobs repeatedly within the same folder due to files that Abaqus generates there. Also, the latest version of Abaci now supports launching jobs via SLURM and handles all this complication behind the scenes.